CONTRIBUTING.md

```diff
@@ -393,6 +393,66 @@ Once the environment is set up, run the network tests using the following comman
 pnpm test:network
 ```
 
+# Transaction Time Measure
+
+A script designed to run locally is available to measure the time required to submit and process different types of transactions on a specified network.
+
+The script code is located at: `packages/fuel-gauge/scripts/latency-detection/main.ts`.
+
+### Setup Instructions
+
+Before running the tests, you need to configure the `.env` file:
+
+1. Copy the `.env.example` file:
+
+```sh
+cp .env.example .env
+```
+
+2. Set the values for the following environment variables in the `.env` file:
+
+```env
+PERFORMANCE_ANALYSIS_TEST_URL=https://testnet.fuel.network/v1/graphql
+PERFORMANCE_ANALYSIS_PVT_KEY=...
+PERFORMANCE_ANALYSIS_CONTRACT_ADDRESS=...
+```
+
+- `PERFORMANCE_ANALYSIS_TEST_URL`: The URL of which network the test should run (e.g., Fuel Testnet endpoint).
+- `PERFORMANCE_ANALYSIS_PVT_KEY`: Your private key for the network.
+- `PERFORMANCE_ANALYSIS_CONTRACT_ADDRESS`: The address of the contract used by the script. If this variable is left empty, the script will deploy the contract before running the time measurement tests. The deployed contract address will be logged, allowing you to add it here for subsequent test runs to avoid re-deployment.
+
+### Running the Test Suite
+
+Once the environment is set up, run the network tests using the following command:
+
+```sh
+pnpm tx:pef
+```
+
+### Output and Results
+
+The test results are saved in the snapshots directory as a CSV file. The filename follows a timestamp format, such as:
```

Comment:
To quick questions:
1. Should this PR add the first version of this snapshot?
2. I know I thought CSV first, but - have you considered JSON?

With things relatively stable, JSON could make it easier to begin automating this process in the future, integrate it with CI, and allow only a certain threshold of deviation from the average, etc.

Reply:
Great, let's go then. Thanks. 🚀

Reply:
https://github.com/FuelLabs/fuels-ts/pull/3583/commits/84632fbe28d8c7812f2d9d48de921b301ce59cca